ddtrace/contrib/django/patch.py

```diff
@@ -189,12 +189,16 @@ def traced_cache(django, pin, func, instance, args, kwargs):
                 db.ROWCOUNT, sum(1 for doc in result if doc) if result and isinstance(result, Iterable) else 0
             )
         elif command_name == "get":
-            # if valid result and check for special case for Django~3.0 that returns an empty Sentinel object as
-            # missing key
-            if result is not None and result != getattr(instance, "_missing_key", None):
-                span.set_metric(db.ROWCOUNT, 1)
-            # else result is invalid or None, set row count to 0
-            else:
+            try:
+                if result is None:
+                    span.set_metric(db.ROWCOUNT, 0)
+                # check also for special case for Django~3.0 that returns an empty Sentinel object for empty results
+                # also check if result is Iterable first since some iterables return ambiguous truth results with ``==``
+                elif not isinstance(result, Iterable) and result == getattr(instance, "_missing_key", None):
```

Comment:
can we merge this with the previous if case since we expect to do the same thing in both cases? i.e.
```python
if result is None or (not isinstance(result. Iterable) and result == getattr(instance, "_missing_key", None):
```

Reply:
Yup, good catch